src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/LaunchSettingsProvider.cs

```diff
@@ -1052,6 +1035,33 @@ internal async Task<string> GetLaunchSettingsFilePathNoCacheAsync()
             return Path.Combine(folder, LaunchSettingsFilename);
         }
 
+        /// <summary>
+        /// Helper retrieves the current snapshot and if there were errors in the launchsettings.json file
+        /// or there isn't a snapshot, it throws an error. There should always be a snapshot of some kind returned.
+        /// </summary>
+        private async Task<ILaunchSettings> GetSnapshotThrowIfErrorsAsync()
+        {
+            ILaunchSettings? currentSettings = await WaitForFirstSnapshot(Timeout.Infinite);
+            Assumes.NotNull(currentSettings);
```

Comment:
This requires the caller to know that the value is non-null when an infinite timeout is passed. I think we have this in a few places. It'd be nice to replace this with a call to a new overload that doesn't accept a timeout and returns a non-null value.

Reply:
Agreed. I think that work would be related to #7056.